src/mlpack/methods/kernel_svm/kernel_svm.hpp

```diff
@@ -0,0 +1,231 @@
+/**
+ * @file methods/kernel_svm/kernel_svm.hpp
+ * @author Himanshu Pathak
+ *
+ * An implementation of Kernel SVM.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_KERNEL_SVM_KERNEL_SVM_HPP
+#define MLPACK_METHODS_KERNEL_SVM_KERNEL_SVM_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <ensmallen.hpp>
+#include <string>
+
+
+namespace mlpack {
+namespace svm {
+
+/**
+ * The KernelSVM class implements an smo algorithm for support vector machine
+ * model, and supports training with multiple non-linear and linear kenel.
+ * The class supports different observation types via the MatType template
+ * parameter; for instance, support vector classification can be performed
+ * on sparse datasets by specifying arma::sp_mat as the MatType parameter.
+ *
+ *
+
+ * @code
+ * @article{Microsoft Research,
+ *   author    = {John C. Platt},
+ *   title     = {Sequential Minimal Optimization:A Fast 
+                  Algorithm for Training Support Vector Machines},
+ *   journal   = {Technical Report MSR-TR-98-14},
+ *   year      = {1998},
+ *   url       = {https://www.microsoft.com/en-us/research
+                  /wp-content/uploads/2016/02/tr-98-14.pdf},
+ * }
+ * @endcode
+ *
+ *
+ * An example on how to use the interface is shown below:
+ *
+ * @code
+ * arma::mat train_data; // Training data matrix.
+ * arma::Row<size_t> labels; // Labels associated with the data.
+ * const size_t inputSize = 1000; // Size of input feature vector.
+ *
+ * // Train the model using default options.
+ * KernelSVM<> svm(train_data, C, kernel_flag, max_iter, tol,
+ *     kernel::Gaussian());
+ *
+ * arma::mat test_data;
+ * arma::Row<size_t> predictions;
+ * lsvm.Classify(test_data, predictions);
+ * @endcode
+ *
+ * @tparam MatType Type of data matrix.
+ * @tparam KernelType Type of kernel used with svm.
+ */
+template <typename MatType = arma::mat,
+          typename KernelType = kernel::LinearKernel>
+class KernelSVM
+{
+ public:
+  /**
+   * Construct the Kernel SVM class with the provided data and labels.
+   *
+   * @param data Input training features. Each column associate with one sample
+   * @param labels Labels associated with the feature data.
+   * @param regularization standard svm regularization parameter.
+   * @param fitKernel true when not using linear kernel.
+   * @param max_iter maximum number of iteration for training.
+   * @param tol tolerance value.
+   * @param kernel kernel type of kernel used with svm.
+   */
+  KernelSVM(const MatType& data,
+            const arma::Row<size_t>& labels,
+            const double regularization = 1.0,
+            const bool fitKernel = false,
```

Comment:
can you remove this fitKernel option? I don't think it should be necessary for the user to specify this.

Reply:
Also can we add an option for `fitIntercept` similar to the one we have in `linearSVM`

Reply:
Hey @saksham189 I have changed everything as suggested but when using `fitkernel` I was able to use this code.
```
    // Giving predictions for linear kernel.
    scores = (data.t() * parameters.t() + intercept).t();
```
So, when giving prediction with `linearkernel` It was faster.
Now I am using
```
    scores = arma::zeros(1, data.n_cols);
    for (size_t i = 0; i < data.n_cols; i++)
    {
      double  prediction = 0;
      for (size_t j = 0; j < trainingData.n_cols; j++)
      {
        if (savedLabels(j) == 0)
          continue;
        prediction = prediction + alpha(j) *
                     savedLabels(j) * kernel.Evaluate(data.col(i),
                     trainingData.col(j));
      }
      scores(i) = prediction + intercept;
    }
```
For Every kernel and doing extra processing in case of linear kernel which not that much efficient . What do you suggest?

Reply:
I think it's fine.